packages/@lwc/engine/src/framework/restrictions.ts

```diff
@@ -382,12 +387,49 @@ function getComponentRestrictionsDescriptors(): PropertyDescriptorMap {
     };
 }
 
-function getLightningElementPrototypeRestrictionsDescriptors(proto: object): PropertyDescriptorMap {
+function getLightningElementPrototypeRestrictionsDescriptors(
+    proto: LightningElement
```

Comment:
I don't think `proto` is a `LightningElement`, it is the prototype of the LightningElement constructor, which is different.